apps/wing/src/commands/test.ts

```diff
@@ -148,68 +150,82 @@ async function testSimulator(synthDir: string) {
-    );
-  }
+async function testTfAws(synthDir: string): Promise<sdk.cloud.TestResult[] | void> {
+  try {
+    if (!isTerraformInstalled(synthDir)) {
+      throw new Error(
+        "Terraform is not installed. Please install Terraform to run tests in the cloud."
+      );
+    }
 
-  await withSpinner("terraform init", () => terraformInit(synthDir));
+    await withSpinner("terraform init", async () => await terraformInit(synthDir));
 
-  await checkTerraformStateIsEmpty(synthDir);
+    if (!(await checkTerraformStateIsEmpty(synthDir))) {
+      console.warn(
+        `Terraform state is not empty. Running \`terraform destroy\` inside ${synthDir} to clean up any previous test runs:`
+      );
+      await withSpinner("terraform destroy", () => terraformDestroy(synthDir));
+    }
+    await withSpinner("terraform apply", () => terraformApply(synthDir));
 
-  await withSpinner("terraform apply", () => terraformApply(synthDir));
+    const [testRunner, tests] = await withSpinner("Setting up test runner...", async () => {
+      const testArns = await terraformOutput(synthDir, ENV_WING_TEST_RUNNER_FUNCTION_ARNS);
+      const testRunner = new TfawsTestRunnerClient(testArns);
 
-  const [testRunner, tests] = await withSpinner("Setting up test runner...", async () => {
-    const testArns = await terraformOutput(synthDir, ENV_WING_TEST_RUNNER_FUNCTION_ARNS);
-    const testRunner = new TfawsTestRunnerClient(testArns);
+      const tests = await testRunner.listTests();
+      return [testRunner, pickOneTestPerEnvironment(tests)];
+    });
 
-    const tests = await testRunner.listTests();
-    return [testRunner, pickOneTestPerEnvironment(tests)];
-  });
+    const results = await withSpinner("Running tests...", async () => {
+      const results = new Array<sdk.cloud.TestResult>();
+      for (const path of tests) {
+        results.push(await testRunner.runTest(path));
+      }
+      return results;
+    });
 
-  const results = await withSpinner("Running tests...", async () => {
-    const results = new Array<sdk.cloud.TestResult>();
-    for (const path of tests) {
-      results.push(await testRunner.runTest(path));
-    }
-    return results;
-  });
+    const testReport = renderTestReport(synthDir, results);
+    console.log(testReport);
 
-  const testReport = renderTestReport(synthDir, results);
-  console.log(testReport);
+    if (testResultsContainsFailure(results)) {
+      console.log("One or more tests failed. Cleaning up resources...");
+    }
 
-  if (testResultsContainsFailure(results)) {
-    console.log("One or more tests failed. Cleaning up resources...");
+    return results;
+  } catch (err) {
+    console.warn((err as Error).message);
+  } finally {
+    cleanup(synthDir);
   }
+}
 
+async function cleanup(synthDir: string) {
   await withSpinner("terraform destroy", () => terraformDestroy(synthDir));
-
-  return results;
+  rmSync(synthDir, { recursive: true, force: true });
```

Comment:
Hmm - I wonder if we are just going to toss away the synth directory anyways, then we might not have a need to use `checkTerraformStateIsEmpty` anymore?

Reply:
Great point! I'll remove it!